payroll/models/hr_payslip.py

```diff
@@ -0,0 +1,803 @@
+            payslip.line_ids.unlink()
+            # write payslip lines
+            number = payslip.number or self.env["ir.sequence"].next_by_code(
+                "salary.slip"
+            )
+            lines = [(0, 0, line) for line in list(payslip.get_lines_dict().values())]
+            payslip.write(
+                {
+                    "line_ids": lines,
+                    "number": number,
+                    "state": "verify",
+                    "compute_date": fields.Date.today(),
+                }
+            )
+        return True
+
+    @api.model
+    def get_worked_day_lines(self, contracts, date_from, date_to):
+        """
+        @param contracts: Browse record of contracts
+        @return: returns a list of dict containing the input that should be
+        applied for the given contract between date_from and date_to
+        """
+        res = []
+        for contract in contracts.filtered(
+            lambda contract: contract.resource_calendar_id
+        ):
+            day_from = datetime.combine(date_from, time.min)
+            day_to = datetime.combine(date_to, time.max)
+            day_contract_start = datetime.combine(contract.date_start, time.min)
+            # Support for the hr_public_holidays module.
+            contract = contract.with_context(
+                employee_id=self.employee_id.id, exclude_public_holidays=True
+            )
+            # only use payslip day_from if it's greather than contract start date
+            if day_from < day_contract_start:
+                day_from = day_contract_start
+            # == compute leave days == #
+            leaves = self._compute_leave_days(contract, day_from, day_to)
+            res.extend(leaves)
+            # == compute worked days == #
+            attendances = self._compute_worked_days(contract, day_from, day_to)
+            res.append(attendances)
+        return res
+
+    def _compute_leave_days(self, contract, day_from, day_to):
+        """
+        Leave days computation
+        @return: returns a list containing the leave inputs for the period
+        of the payslip. One record per leave type.
+        """
+        leaves_positive = (
+            self.env["ir.config_parameter"].sudo().get_param("payroll.leaves_positive")
+        )
+        leaves = {}
+        calendar = contract.resource_calendar_id
+        tz = timezone(calendar.tz)
+        day_leave_intervals = contract.employee_id.list_leaves(
+            day_from, day_to, calendar=contract.resource_calendar_id
+        )
+        for day, hours, leave in day_leave_intervals:
+            holiday = leave[:1].holiday_id
+            current_leave_struct = leaves.setdefault(
+                holiday.holiday_status_id,
+                {
+                    "name": holiday.holiday_status_id.name or _("Global Leaves"),
+                    "sequence": 5,
+                    "code": holiday.holiday_status_id.name or "GLOBAL",
```

Comment:
@mtelahun Okay so here you continue using name. Also i think you have to add the new file adding the field to hr.leave.type.

I would prefer we include this change in this PR because its a crucial functionality of the module.

Reply:
Done